docs/architecture/state.md

```diff
@@ -1,91 +1,111 @@
+- **Private accounts:** where only the hash of the account is stored on-chain.
 
-Private accounts significantly reduce the storage overhead for nodes. A private account contributes only $40$ bytes to the global state ($8$ bytes account ID + $32$ bytes account hash). Or, said another way, 1 billion private accounts takes up only $40$ GB of state.
+Private accounts significantly reduce storage overhead. A private account contributes only $40$ bytes to the global state ($8$ bytes for the account ID + $32$ bytes for the account hash). For example, 1 billion private accounts take up only $40$ GB of state.
+
+The storage contribution of a public account depends on the amount of data it stores.
 
 > **Warning**
->  Losing the state of a private account means loss of funds in a similar manner as a loss of a private key - as the user won't be able to execute transactions. This problem can be mitigated by storing encrypted account state in the cloud or backing it up somewhere else. Unlike storing private keys in the cloud, this does not compromise privacy or account security.
->
->  In the future we hope to enable encrypted accounts where the account data is stored on-chain but in an encrypted format. This is especially interesting for shared accounts like advanced multi-sig wallets.
+> - In Miden, when the user is the custodian of their account state (in the case of a private account), losing this state amounts to losing their funds, similar to losing a private key.
+
+### Notes database
+
+As described in the [notes section](notes.md), there are two types of notes:
 
-### Note database
+- **Public notes:** where the entire note content is stored on-chain.
+- **Private notes:** where only the note’s hash is stored on-chain.
 
-Notes are recorded in an append-only accumulator, a [Merkle Mountain Range](https://github.com/opentimestamps/opentimestamps-server/blob/master/doc/merkle-mountain-range.md). Each leaf is a block header which contains the commitment to all notes created in that block. The commitment is a Sparse Merkle Tree of all the notes in a block. The size of the Merkle Mountain Range grows logarithmically with the number of items in it.
+Private notes greatly reduce storage requirements and thus result in lower fees. They add only $64$ bytes to the state ($32$ bytes when produced and $32$ bytes when consumed).
+
+At high throughput (e.g., 1K TPS), the note database could grow by about 1TB/year. However, only unconsumed public notes and enough information to construct membership proofs must be stored explicitly. Private notes, as well as consumed public notes, can be discarded. This solves the issue of infinitely growing note databases.
 
 ![Architecture core concepts](../img/architecture/state/note-db.png)
 
-As described in [the notes section](notes.md), there are two types of notes:
+### Nullifiers database
+
+Nullifiers map one-to-one to existing notes, tracking whether a note has been consumed.
```

Comment:
```suggestion
Every note has a nullifier [LINK TO NOTE SECTION]. The note has been consumed if the nullifier is recorded in the database. 
```

Simpler wording?

Reply:
Maybe also add that the nullifier db prevents double spending. Notes can only be consumed once.

Reply:
Reworded and added.